Added a utility function to help encapsulate the dependency on generator escapePluginId

mcilroyc opened the pull request:
Some light encoding is performed by generator on plugin *names* in order to safely store their preferences in photoshop.  Since consumers of generator-connection will likely want to fetch those preferences  (eg, the port number), this utility function helps (I think) formalize that soft dependency.  Thoughts @iwehrman?

iwehrman commented on main.js:
```diff
@@ -23,11 +23,32 @@
+     * The replacement logic must match the way generator-core
+     * See: generator's escapePluginId() function
+     *
+     * @type {?string}
```
`@param`? Also, is there any reason for this function to accept null?

iwehrman replied:
And `@return`? Or is this a thing where you're matching some other old syntax found in this repo?

iwehrman commented:
Besides the nits, seems fine.

mcilroyc commented:
I cleaned up the JSDoc, and simplified the logic slightly.  To answer your question, there is no reason this should accept null.  It does seem reasonable for the function to just return null if it was supplied, so I left that quick check in there - but if think it should be removed, I'll do it.

volfied commented:
Per stand-up, taking this and merging it. LGTM
